src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -190,15 +190,27 @@ abstract class ComputeAdviceTask @Inject constructor(
 
       output.bufferWriteJson(projectAdvice)
       // These must be transformed so that the Coordinates are Strings for serialization
-      dependencyUsagesOut.bufferWriteJsonMap(dependencyUsages.toStringCoordinates())
-      annotationProcessorUsagesOut.bufferWriteJsonMap(annotationProcessorUsages.toStringCoordinates())
+      dependencyUsagesOut.bufferWriteJsonMap(dependencyUsages.toStringCoordinates(buildPath))
+      annotationProcessorUsagesOut.bufferWriteJsonMap(annotationProcessorUsages.toStringCoordinates(buildPath))
       // TODO consider centralizing this logic in a separate PR
       bundleTraces.bufferWriteJsonSet(dependencyAdviceBuilder.bundledTraces)
     }
 
-    private fun Map<Coordinates, Set<Usage>>.toStringCoordinates(): Map<String, Set<Usage>> = map { (key, value) ->
-      key.gav() to value
-    }.toMap()
+    private fun Map<Coordinates, Set<Usage>>.toStringCoordinates(buildPath: String): Map<String, Set<Usage>> =
+      map { (key, value) ->
+        toCoordinatesKey(key, buildPath) to value
+      }.toMap()
+
+    private fun toCoordinatesKey(coordinates: Coordinates, buildPath: String) =
+      coordinates.gav() + when {
+        // If the coordinates point at a project in the current build, there are two notations to address it:
+        // 1) 'group:name' coordinates, which is the main representation we aim to use everywhere (see IncludedBuildCoordinates)
+        // 2) The project path notation starting with a ':' (:project-name)
+        // To allow notation 2) to address a project in the 'reasons' task, we add this coordinate to the key as well if applicable (separated by '|')
+        coordinates is IncludedBuildCoordinates && coordinates.resolvedProject.buildPath == buildPath -> "|${coordinates.resolvedProject.gav()}"
+        else -> ""
```

Comment:
In the `usages-*.json` files, the 'id' is the String key. We somehow need to pass along the alternative ID here. Since I did not want to change the whole model there, I now append the alternative 'id' – the project path – to the key separated by `|`.

Reply:
Can we extract this code into a utility class for better encapsulation and also use by `ReasonTask`? Then it would be clear how the `|` works and who uses it.

Reply:
Moved all functions related to this into `utils/coordinatesStrings.kt`.